File: 02-print-hello-world-loop/py/src/main.py
#!/usr/bin/env python

import argparse
import time

program_version = "Hello World Loop 0.0.1"

#######################################################
#      Main Program
#######################################################
def main():
    parser = argparse.ArgumentParser(description='Prints a specified number of "Hello World" lines.')
    parser.add_argument("-v", "--version", help="Print program version", action="version", version=program_version)
    parser.add_argument("--lines", "-l", help="Number of lines to print, defaults to 100", default=100, type=int)
    args = parser.parse_args()

    st = time.time()
    for i in range(args.lines):
        print("Hello World")
    et = time.time()

    print("Time taken to execute {} prints: {}".format(args.lines, et - st))

if __name__ == "__main__":
    main()

File: 02-print-hello-world-loop/js/main.js
#!/usr/bin/env node

const program = require('commander');

program
.description('Command Line Interface Hello World')
.usage('[-h] [-v] [-l LINES]')
.option('-l, --lines <int>', 'Repeat execution "Hello World" times.', 100, parseInt)
.version('Hello World Loop 1.0.0', '-v, --version')
.parse(process.argv);

const options = program.opts();

/***************************************************************************
 *      Main Program
 ***************************************************************************/
function main() {
    if (options == undefined || !options.lines == undefined) {
        console.log(`Error parsing options`);
        return 0;
    }

    const st = new Date();

    for (let i = 0; i < options.lines; i++) {
        console.log(`Hello World`);
    }

    const et = new Date();
    const tt = (et.getTime() - st.getTime()) / 1000;

    console.log(`Time taken to execute ${options.lines} prints: ${tt}`);
}

main();
